### DISCUSSION TITLE
When in bridged AP mode, should the MAC addresses of the WiFi clients connected to the bridge be shown under connected devices?

### DISCUSSION THREAD

**Discussion by tehbra1n** [RaspAP/raspap-webgui#1694] [Q&A] [answered]
The bridge is working as expected, but I assumed that the dashboard would show the MAC addresses of the wireless clients connected to the bridge, but but it only shows the info the screenshot below.

![image](https://github.com/user-attachments/assets/94783a4f-390a-4ecb-a524-a07b66ea9ae7)

**Comment by billz** [comment]
In bridged AP mode, [all DHCP functions are delegated to your upstream router](https://github.com/RaspAP/raspap-webgui?tab=readme-ov-file#bridged-ap). RaspAP has no knowledge whatsoever of client leases in this configuration.

**Reply by tehbra1n** [reply]
I understand that in bridge mode the AP just working on layer 2 and its not concerned with IP addresses.  The wireless clients are still associating with the AP, and the AP should know the MAC addresses of any clients connected to it.  It seemed like that list of connected MACs could be displayed in the dashboard.

So if the connected devices section is just meant to show the IP addresses the AP handed out via DHCP when in router mode, and is meant to be empty when in bridged AP mode, that answers my question.

**Reply by billz** [reply, marked as answer]
A list of MAC addresses belonging to clients associated with the AP should appear here. Here is the [method used to create the client list](https://github.com/RaspAP/raspap-webgui/blob/a86c637f049963003a74a2da431743b72b97e3cc/includes/dashboard.php#L193). e.g.:

```
iw dev wlan0 station dump | grep -oE "([[:xdigit:]]{2}:){5}[[:xdigit:]]{2}"
```
Substitute `wlan0` for your AP interface.

If I connect a client to a bridged AP it returns the client MAC.
